osidb/api_views.py

```diff
@@ -738,6 +739,80 @@ class AffectView(SubFlawViewDestroyMixin, ModelViewSet):
     http_method_names = get_valid_http_methods(ModelViewSet)
     permission_classes = [IsAuthenticatedOrReadOnly]
 
+    @extend_schema(
+        request=AffectSerializer(many=True),
+        responses=AffectBulkPutResponseSerializer,
+    )
+    @action(methods=["PUT"], detail=False, url_path="bulk")
+    def put(self, request, *args, **kwargs):
+        """
+        Bulk update endpoint. Expects a list of dict Affect objects.
+        """
+
+        bz_api_key = request.META.get("HTTP_BUGZILLA_API_KEY")
+        if not bz_api_key:
+            raise ValidationError({"Bugzilla-Api-Key": "This HTTP header is required."})
+
+        # TODO sometime: Some of these actions probably belong to another layer, perhaps serializer.
```

Comment:
I agree with moving this logic to a serializer. See https://www.django-rest-framework.org/api-guide/serializers/#customizing-listserializer-behavior

Reply:
I will check the rest later, as I do not have enough time today. My apologies.

Reply:
I will not review/block this PR since it was decided to stick with the current version at today's meeting.